### ISSUE TITLE
[ARCHETYPE-521] Cannot specify remote repository for generate project from archetype

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archetype#517]
**[Dmytro Sylaiev](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=limych)** opened **[ARCHETYPE-521](https://issues.apache.org/jira/browse/ARCHETYPE-521?redirect=false)** and commented

I wanted to create new project using `mvn archetype:generate` command.
Desired archetype was absent in the central repo, so I tried to use `-DarchetypeRepository=repo.url` param. But this param is out of date in 3.0.0 version of plugin.
So I needed to use  `mvn org.apache.maven.plugins:maven-archetype-plugin:2.4:generate` command.

Question: How can I specify specific repository for generating project (without editing `settings.xml` file)?
(`-DarchetypeCatalog=repo.url` doesn't work too)

---

**Affects:** 3.0.0

**Issue Links:**
- [ARCHETYPE-438](https://issues.apache.org/jira/browse/ARCHETYPE-438) Remove support for specifying catalogs through direct URLs
 (_**"depends upon"**_)
- [ARCHETYPE-439](https://issues.apache.org/jira/browse/ARCHETYPE-439) Remove archetypeRepository param
 (_**"depends upon"**_)

1 votes, 6 watchers

**Comment by jira-importer** [comment]
**[Nathan Thomas](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nthomas)** commented

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte) There doesn't seem to be any documentation surrounding specifying the repository (id: archetype) in the settings.xml, I did the obvious and under my active profile created a repository with the id set to archetype and the url pointed to my archetype-catalog.xml, when that didn't work I tried just pointing to my repository, both were not successful. What am I missing here?  Any example or documentation I can see related to this?
Is there a parameter similar to \<releases> or \<snapshot> that needs to be enabled?
